optuna/visualization/matplotlib/_optimization_history.py

```diff
@@ -1,11 +1,79 @@
 from optuna._experimental import experimental
+from optuna.logging import get_logger
+from optuna.study import Study
+from optuna.study import StudyDirection
+from optuna.trial import TrialState
 from optuna.visualization.matplotlib._matplotlib_imports import _imports
 
 
 if _imports.is_successful():
     from optuna.visualization.matplotlib._matplotlib_imports import Axes
+    from optuna.visualization.matplotlib._matplotlib_imports import plt
+
+_logger = get_logger(__name__)
 
 
 @experimental("2.2.0")
-def plot_optimization_history() -> Axes:
-    raise NotImplementedError("To be implemented soon.")
+def plot_optimization_history(study: Study) -> "Axes":
+    """Plot optimization history of all trials in a study with Matplotlib.
+
+    .. seealso::
+        Please refer to :func:`optuna.visualization.plot_optimization_history` for an example.
+
+    Args:
+        study:
+            A :class:`~optuna.study.Study` object whose trials are plotted for their objective
+            values.
+
+    Returns:
+        A :class:`matplotlib.axes.Axes` object.
+    """
+
+    _imports.check()
+    return _get_optimization_history_plot(study)
+
+
+def _get_optimization_history_plot(study: Study) -> "Axes":
+
+    # Set up the graph style.
+    plt.style.use("ggplot")  # Use ggplot style sheet for similar outputs to plotly.
+    fig, ax = plt.subplots()
```

Comment:
A nit, but I guess `fig` is not used in this function.
```suggestion
    _, ax = plt.subplots()
```

Reply:
Thank you for reviewing! Removed it.

Reply:
Thank you for your swift action. LGTM again.